tokio-timer/src/interval.rs

```diff
@@ -0,0 +1,55 @@
+use Sleep;
+
+use futures::{Future, Stream, Poll};
+
+use std::time::{Instant, Duration};
+
+/// A stream representing notifications at fixed interval
+#[derive(Debug)]
+pub struct Interval {
+    sleep: Sleep,
+    duration: Duration,
+}
+
+impl Interval {
+    /// Create a new `Interval` that starts at `at` and yields every `duration`
+    /// interval after that.
+    ///
+    /// The `duration` argument must be a non-zero duration.
+    ///
+    /// # Panics
+    ///
+    /// This function panics if `duration` is zero.
+    pub fn new(at: Instant, duration: Duration) -> Interval {
```

Comment:
Would it not be more common to just want `at` to always be `Instant::now()`?

Reply:
Probably, but my thought was to leave out any call to `Instant::now()`. At the point where `new` i called, one does not have access to the source of `now`. There might be a way to access it, but it is not immediately obvious how.